daemon/daemon.go

```diff
@@ -442,7 +442,17 @@ configRetry:
 		log.Infof("Starting the Typha connection")
 		err := typhaConnection.Start(context.Background())
 		if err != nil {
-			log.WithError(err).Fatal("Failed to connect to Typha")
+			retry := 0
+			for err != nil && retry < 10 {
+				// Set Ready and Live to false
+				healthAggregator.Report(healthName, &health.HealthReport{Live: false, Ready: false})
+				err = typhaConnection.Start(context.Background())
+				log.WithError(err).Warn("Retrying to start Typha")
+				retry++
+			}
+			if err != nil && retry > 10 {
```

Comment:
This condition will never fire; retry will be exactly 10 after the loop finishes.

Reply:
Ack